### PR TITLE
BATS tests: make more robust

### PR DESCRIPTION
**Pull request by edsantiago**
Add a --retry flag to run_buildah; intended for tests which
really need to pull an image, and can't rely on prefetch().
It will try a failed buildah command three times, waiting
thirty seconds between retries.

This is imperfect: there's no way to specify the number of
retries, no way to specify the timeout, and no way to
combine --retry with an expected (i.e. nonzero) exit
status. (FWIW I can't think of any possible use for that).
It is, though, quick, minimal-impact, easy to remember
and use.

Add --retry option to various tests in pull.bats and
registries.bats.

And, while I'm at it, add expect_output checks to many
of the new encryption checks in pull.bats and from.bats.
This actually caught a bug in a test, a check that was
failing (expected) but for the wrong reason (missing
file, not wrong key). Have I mentioned, lately, that
tests should check error message strings, not just
exit status?

Fixes: #2473

Signed-off-by: Ed Santiago <santiago@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by edsantiago** [comment]
Quick note: I tested `--retry` by misspelling "fedora" in one `registries.bats` test. Although not an actual network flake, the test did take a long time, eventually failed, and displayed three repeat commands (each with error status, two of them with a retry message).

File: tests/pull.bats [hunk 1]
```diff
@@ -209,17 +218,23 @@ load helpers
...
   run_buildah from --quiet --pull=false --signature-policy ${TESTSDIR}/policy.json busybox
   cid=$output
   run_buildah commit --iidfile /dev/null --tls-verify=false --creds testuser:testpassword --signature-policy ${TESTSDIR}/policy.json --encryption-key jwe:${TESTDIR}/tmp/mykey.pub -q $cid docker://localhost:5000/buildah/busybox_encrypted:latest
 
   # Try to pull encrypted image without key should fail
   run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword docker://localhost:5000/buildah/busybox_encrypted:latest
+  expect_output --substring "Error decrypting layer .* missing private key needed for decryption"
+
   # Try to pull encrypted image with wrong key should fail
   run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json --tls-verify=false --creds testuser:testpassword --decryption-key ${TESTDIR}/tmp/mykey2.pem docker://localhost:5000/buildah/busybox_encrypted:latest
```
Comment (TomSweeneyRedHat) [inline tests/pull.bats:232]: nice catch

**Comment by TomSweeneyRedHat** [comment]
LGTM
thx @edsantiago 
Happy green test buttons too....

**Comment by rhatdan** [comment]
bors retry

**Pull request merged**